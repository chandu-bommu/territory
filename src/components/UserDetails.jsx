import React, { Component } from 'react';
import '../style/App.css';
import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { TextField } from 'office-ui-fabric-react/lib/TextField';


class UserDetails extends React.Component{

  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <div className="userForm">
          <TextField label='Required' required={ true } />
          <PrimaryButton data-automation-id='getDetailsButton' text='Get Details' />
        </div>
        <div className="userCard">
          <div className="userImage">
          </div>
          <div className="userInfo">
          </div>
        </div>
      </div>
    );
  }
}

export default UserDetails;
