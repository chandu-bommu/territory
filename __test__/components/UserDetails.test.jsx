import React from 'react';
import UserDetails from '../../src/components/UserDetails.jsx';
import { shallow, mount } from 'enzyme';

describe('UserDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UserDetails />);
    expect(wrapper).toMatchSnapshot();
  });
});
