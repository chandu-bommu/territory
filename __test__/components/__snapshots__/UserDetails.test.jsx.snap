// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="appComponent"
  >
    <div
      className="userDetails"
    >
      <div
        className="header"
      >
        User Details
      </div>
      <table>
        <tbody>
          <tr>
            <td
              style={
                Object {
                  "width": "360px",
                }
              }
            >
              <TextField
                autoAdjustHeight={false}
                borderless={false}
                deferredValidationTime={200}
                errorMessage=""
                id="userId"
                label="Enter User ID"
                multiline={false}
                onBeforeChange={[Function]}
                onChanged={[Function]}
                onGetErrorMessage={[Function]}
                onNotifyValidationResult={[Function]}
                placeholder="Enter UserID"
                resizable={true}
                underlined={false}
                validateOnFocusIn={false}
                validateOnFocusOut={false}
                validateOnLoad={true}
                value={undefined}
              />
            </td>
            <td
              style={
                Object {
                  "paddingTop": "30px",
                }
              }
            >
              <CustomizedPrimaryButton
                data-automation-id="getDetailsButton"
                onClick={[Function]}
                text="Get Details"
              />
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="userCard"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="userImage"
        >
          <img
            alt="avtar"
            src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
            width="190"
          />
        </div>
        <div
          className="userInfo"
        >
          <span
            id="idSpan"
          >
            ID: 
            
          </span>
          <span
            id="nameSpan"
          >
            Name: 
            
          </span>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="appComponent"
    >
      <div
        className="userDetails"
      >
        <div
          className="header"
        >
          User Details
        </div>
        <table>
          <tbody>
            <tr>
              <td
                style={
                  Object {
                    "width": "360px",
                  }
                }
              >
                <TextField
                  autoAdjustHeight={false}
                  borderless={false}
                  deferredValidationTime={200}
                  errorMessage=""
                  id="userId"
                  label="Enter User ID"
                  multiline={false}
                  onBeforeChange={[Function]}
                  onChanged={[Function]}
                  onGetErrorMessage={[Function]}
                  onNotifyValidationResult={[Function]}
                  placeholder="Enter UserID"
                  resizable={true}
                  underlined={false}
                  validateOnFocusIn={false}
                  validateOnFocusOut={false}
                  validateOnLoad={true}
                  value={undefined}
                />
              </td>
              <td
                style={
                  Object {
                    "paddingTop": "30px",
                  }
                }
              >
                <CustomizedPrimaryButton
                  data-automation-id="getDetailsButton"
                  onClick={[Function]}
                  text="Get Details"
                />
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="userCard"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="userImage"
          >
            <img
              alt="avtar"
              src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
              width="190"
            />
          </div>
          <div
            className="userInfo"
          >
            <span
              id="idSpan"
            >
              ID: 
              
            </span>
            <span
              id="nameSpan"
            >
              Name: 
              
            </span>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <UserDetails />,
    "_forcedUpdate": false,
    "_instance": UserDetails {
      "_onChanged": [Function],
      "context": Object {},
      "getUserDetails": [Function],
      "props": Object {},
      "refs": Object {},
      "render": [Function],
      "state": Object {
        "display": "none",
        "imgSrc": "data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=",
        "infoStatus": "",
        "userIdNumber": "",
        "userName": "",
        "value": undefined,
      },
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="appComponent"
    >
      <div
        className="userDetails"
      >
        <div
          className="header"
        >
          User Details
        </div>
        <table>
          <tbody>
            <tr>
              <td
                style={
                  Object {
                    "width": "360px",
                  }
                }
              >
                <TextField
                  autoAdjustHeight={false}
                  borderless={false}
                  deferredValidationTime={200}
                  errorMessage=""
                  id="userId"
                  label="Enter User ID"
                  multiline={false}
                  onBeforeChange={[Function]}
                  onChanged={[Function]}
                  onGetErrorMessage={[Function]}
                  onNotifyValidationResult={[Function]}
                  placeholder="Enter UserID"
                  resizable={true}
                  underlined={false}
                  validateOnFocusIn={false}
                  validateOnFocusOut={false}
                  validateOnLoad={true}
                  value={undefined}
                />
              </td>
              <td
                style={
                  Object {
                    "paddingTop": "30px",
                  }
                }
              >
                <CustomizedPrimaryButton
                  data-automation-id="getDetailsButton"
                  onClick={[Function]}
                  text="Get Details"
                />
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="userCard"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="userImage"
          >
            <img
              alt="avtar"
              src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
              width="190"
            />
          </div>
          <div
            className="userInfo"
          >
            <span
              id="idSpan"
            >
              ID: 
              
            </span>
            <span
              id="nameSpan"
            >
              Name: 
              
            </span>
          </div>
        </div>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserDetails />,
}
`;
